data/data/agent/systemd/units/assisted-service.service.template

```diff
@@ -11,7 +11,7 @@ Environment=PODMAN_SYSTEMD_UNIT=%n
 Restart=on-failure
 TimeoutStopSec=300
 ExecStartPre=/bin/rm -f %t/%n.ctr-id
-ExecStart=/usr/bin/podman run --cidfile=%t/%n.ctr-id --cgroups=no-conmon --log-driver=journald --rm --pod-id-file=%t/assisted-service-pod.pod-id --sdnotify=conmon --replace -d --name=service -v /opt/agent/tls:/opt/agent/tls:z {{.MirrorRegistriesMount}} {{.CaBundleMount}} --env-file=/usr/local/share/assisted-service/assisted-service.env --env-file=/usr/local/share/assisted-service/images.env --env-file=/etc/assisted-service/node0 quay.io/edge-infrastructure/assisted-service:latest
+ExecStart=/usr/bin/podman run --cidfile=%t/%n.ctr-id --cgroups=no-conmon --log-driver=journald --rm --pod-id-file=%t/assisted-service-pod.pod-id --sdnotify=conmon --replace -d --name=service -v /opt/agent/tls:/opt/agent/tls:z {{ if .ReleaseImageMirror }}-v /etc/containers:/etc/containers{{ end }} -v /etc/pki/ca-trust:/etc/pki/ca-trust --env-file=/usr/local/share/assisted-service/assisted-service.env --env-file=/usr/local/share/assisted-service/images.env --env-file=/etc/assisted-service/node0 quay.io/edge-infrastructure/assisted-service:latest
```

Comment:
The conditional on ReleaseImageMirror isn't working as expected. When I tested the disconnected case using dev-scripts ("export MIRROR_IMAGES=true" in config_xxx.sh), ReleaseImageMirror has empty string as value. /etc/containers doesn't get mounted and the install-config.yaml for the cluster to be installed is missing additionalTrustBundle and imageContentSources.

Reply:
Yes, good find. The way it worked previously is MirrorRegistriesMount is set up based on the existence of the input registries.conf, not on the ReleaseImageMirror - https://github.com/openshift/installer/blob/89ebbd566cb7d7479e828f87f0248915f0eea946/pkg/asset/agent/image/ignition.go#L137

As you noted below, in dev-scripts the ReleaseImage is to the override location, i.e. the mirrored location.